extensions/blocks/simple-payments/edit.js

```diff
@@ -90,34 +90,26 @@ class SimplePaymentsEdit extends Component {
 		 * When subsequent saves occur, we should avoid injecting attributes so that we do not
 		 * overwrite changes that the user has made with stale state from the previous save.
 		 */
-		if ( ! this.shouldInjectPaymentAttributes ) {
+
+		const { simplePayment } = this.props;
+		if ( ! this.shouldInjectPaymentAttributes || ! Object.keys( simplePayment ).length ) {
```

Comment:
This can be simplified since empty objects are _falsy_:

<img width="73" alt="Screen Shot 2019-03-28 at 14 29 52" src="https://user-images.githubusercontent.com/1233880/55132503-fa09be00-5165-11e9-8209-b4ca0c0c9cf0.png">
```suggestion
		if ( ! this.shouldInjectPaymentAttributes || ! simplePayment ) {
```

Reply:
Thanks. Updating.

Reply:
updated and rebased